### PR TITLE
Improved cmake build instructions

### LINKED ISSUE #35: CMake clean-up

### PR DESCRIPTION
**Pull request by mkaring**
The server is not properly building with cmake.

fixes #35

### CODE REVIEW

**Review by vilarion — CHANGES_REQUESTED** [review]

**Comment by vilarion** [comment]
CMake Error at /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PostgreSQL (missing: PostgreSQL_TYPE_INCLUDE_DIR) (found
  version "11.9 (Debian 11.9-0+deb10u1)")
Call Stack (most recent call first):
  /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.13/Modules/FindPostgreSQL.cmake:175 (find_package_handle_standard_args)
  cmake/Modules/FindPqxx.cmake:2 (find_package)
  src/CMakeLists.txt:4 (find_package)

**Comment by mkaring** [comment]
@vilarion: Your environment is likely lacking the postgresql development dependency: `postgresql-server-dev-all`

**Comment by vilarion** [comment]
postgresql-server-dev-all should not be a required dependency, we are not working with libpq directly.

**Comment by vilarion** [comment]
libpq-dev is already required by libpqxx-dev, but we don't need it directly.

**Comment by mkaring** [comment]
The `FindPqxx.cmake` script was wrong in depending on PostgreSQL.

File: cmake/Modules/FindLuabind.cmake [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# This module defines:
+#   LUABIND_FOUND - if false, do not try to link to Luabind
+#   LUABIND_LIBRARIES - luabind libary and lua library
+#   LUABIND_INCLUDE_DIRS - where to find luabind/luabind.hpp
+#   LUABIND_VERSION_STRING - the version of Luabind found
+#   LUABIND_VERSION_MAJOR - the major version of Luabind
+#   LUABIND_VERSION_MINOR - the minor version of Luabind
+#   LUABIND_VERSION_PATCH - the patch version of Luabind
+
+find_package( Lua )
```
[the remark this answers is not in the document]
Comment (mkaring) [reply cmake/Modules/FindLuabind.cmake:11]: Done in the server `CMakeLists.txt`

**Review by vilarion — APPROVED** [review]

**Comment by vilarion** [comment]
Rebased and merged.

**Pull request closed without merging**